### ISSUE TITLE
For calculating QNM of Schwarzschild BH with (l,m,n) = (3,3,0) the value of QNF i wrong

### ISSUE THREAD

**Issue by AfrozSaqlain** [raimonluna/MachineLearningForStrongGravity#1]
I ran the code for calculating QNM frequency for Schwarzschild BH with (l,m,n) = (3,3,0) and the final value which I get for omega, has its imaginary component way off as compared to the actual value. I will attach the image for you to look over.

I also read your paper and there it's mentioned that you got the correct value for QNF for the (3,3,0) case, but the code on GitHub is not giving the desired result.

Please look into this issue.

![Image](https://github.com/user-attachments/assets/ce8d21c1-3924-4a6b-b830-647b49202ed3)

**Comment by raimonluna** [comment]
Use the paper specifications for the radial neural network:

        self.netR = nn.Sequential(
            nn.Linear(1, 200),
            nn.Tanh(),
            nn.Linear(200, 200),
            nn.Tanh(),
            nn.Linear(200, 200),
            nn.Tanh(),
            nn.Linear(200, 2),
        )

And in this case it works better with a slightly larger initial seed, such as :

        self.w   = nn.Parameter(torch.tensor(1.0 - 0.1j), requires_grad = True)

The final result should be close to w = 1.2016 − 0.1852i. I hope this helps!

**Comment by AfrozSaqlain** [comment]
Thank you for your response. 

Regarding the initial omega value estimations within the PINN, could we explore methodologies to eliminate the reliance on prior knowledge-based guesses while maintaining convergence?

Similarly, for higher harmonics, our current approach involves providing omega values proximate to the true values to achieve convergence. Is it feasible to employ gradient descent to initially determine omega, followed by gradient ascent to identify subsequent loss function minima corresponding to higher overtones?

Alternatively, could we investigate a method to exclude the minimal loss function region (obtained during QNM frequency calculation for n=0) and train the model to discover alternative solutions?

**Comment by raimonluna** [comment]
There are ways in the literature that explore criteria to decide which initial seeds will converge to solutions. In any case, if you are interested in high precision overtones, I would recommend other methods such as Leaver's continued fraction (if the background is analytical) or the one shown here: https://github.com/raimonluna/LecturesOnSpectralMethods/blob/main/Lecture7_Multidomain_Quasinormal_Modes.ipynb
